### PR TITLE
Finished tests module

### PR DESCRIPTION
**Pull request by VanderY**

### CODE REVIEW

**Comment by NikitaZotov** [comment]
Please, rename commits names connected with merges.

File: sources/bot/modules/chains/survey/student_handlers_chain.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        survey_sheet_name = separated_data[1]
+        data = await state.get_data()
+        tests = data.get("tests")
+
+        with open(f'surveys/{survey_sheet_name}.json', encoding='utf-8') as json_file:
+            survey = json.load(json_file)
+            if tests is None:
+                await state.update_data(tests={"is_finished": False, "answers": {}, "test_name": survey_sheet_name})
+            question_number = int(separated_data[2])
+            # Проверка ответов на правильность
```
Comment (NikitaZotov) [inline sources/bot/modules/chains/survey/student_handlers_chain.py:63]: Please, translate comments in English.

File: sources/bot/modules/chains/survey/student_handlers_chain.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                                                  f"(id:{callback_query.message.chat.id}) "
+                                                  f"passed test")
+                StudentHandlersChain._logger.info(f"Answers: {answers}")
+
+                tests["answers"] = answers
+                await state.update_data(tests=tests)
+
+                await SurveyStudentStates.next()
+                await callback_query.message.edit_text(text=f"Вы прошли тест на {correct_answers}/{len(answers)}")
+                await callback_query.answer()
```
Comment (NikitaZotov) [inline sources/bot/modules/chains/survey/student_handlers_chain.py:106]: I think it is worth breaking this method down into several simpler ones.

File: sources/bot/modules/chains/survey/student_handlers_chain.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    async def ready_to_pass_survey_handler(query: types.CallbackQuery, state: FSMContext):
+        data = await state.get_data()
+
+        await query.message.edit_text("Ожидайте сообщения о начале теста")
+        await SurveyStudentStates.student_ready_to_pass_test.set()
+
+    @staticmethod
+    @Registrar.callback_query_handler(lambda c: c.data, state=SurveyStudentStates.student_ready_to_pass_test)
+    async def passing_test_handler(callback_query: types.CallbackQuery, state: FSMContext):
+        separated_data = callback_query.data.split(";")
```
Comment (NikitaZotov) [inline sources/bot/modules/chains/survey/student_handlers_chain.py:53]: What separated data is? Please, explain it.

File: sources/bot/modules/factory/standard_handlers_factory.py [hunk 4]
```diff
@@ -1,11 +1,12 @@
 """
 Standard bot handlers factory implementation module.
 """
+from ..chains.survey.student_handlers_chain import StudentHandlersChain
+from ..chains.survey.teacher_handlers_chain import SurveyTeacherHandlersChain
 from ...loggers import LogInstaller
 from ...modules.chains.auth.auth_handlers_chain import AuthHandlersChain
 from ...modules.handlers_registrar import HandlersRegistrar
 from ...modules.chains.main.main_handlers_chain import MainHandlersChain
```
Comment (NikitaZotov) [inline sources/bot/modules/factory/standard_handlers_factory.py:9]: Please, make correct order.

File: sources/bot/storage/spreadsheet/tests/tests_spreadsheet_handler.py [hunk 5]
```diff
@@ -1,13 +1,137 @@
...
+                boolean_answer_list.append("Неверно")
+
+        row = [user_data, str(datetime.now())]
+
+        for ans in boolean_answer_list:
+            row.append(ans)
+
+        row.append(f"{correct_answers}/{len(result_list)}")
+
+        self._add_row(test_name + "_result", row, self._current_test_id)
```
Comment (NikitaZotov) [inline sources/bot/storage/spreadsheet/tests/tests_spreadsheet_handler.py:119]: I think it is worth breaking this method down into several simpler ones.

File: sources/bot/storage/spreadsheet/tests/tests_spreadsheet_handler.py [hunk 6]
```diff
@@ -1,13 +1,137 @@
...
 
 
 class TestsSpreadsheetHandler(BaseTestsSpreadsheetHandler):
-    # todo: Implement tests handler
-    def __init__(self, spreadsheet_id: str, token: str):
-        pass
+    def __init__(self, credentials_file_name: str):
+        self._loaded_tests = {}
+        self._current_test_id = ""
+        self._http_auth = ServiceAccountCredentials.from_json_keyfile_name(
```
Comment (NikitaZotov) [inline sources/bot/storage/spreadsheet/tests/tests_spreadsheet_handler.py:18]: We have a class that does this kind of logic. We were talking about it. Please, use it.

File: sources/bot/storage/spreadsheet/tests/tests_spreadsheet_handler.py [hunk 7]
```diff
@@ -1,13 +1,137 @@
...
+        self._save_test(test_name, survey)
+        return test_name, survey
+
+    def _save_test(self, test_name: str, test: list[dict]) -> None:
+        if not test_name == "":
+            current_path = os.path.dirname(sys.argv[0])  # needs to be configured in config.ini
+            path = f'{current_path}\\surveys'
+            os.makedirs(path, exist_ok=True)
+            with open(f'surveys/{test_name}.json', 'w', encoding='utf-8') as f:
+                json.dump(test, f, ensure_ascii=False, indent=4)
```
Comment (NikitaZotov) [inline sources/bot/storage/spreadsheet/tests/tests_spreadsheet_handler.py:91]: Please, move the logic for working with json into a separate class.

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

**Comment by VanderY** [comment]
Made new PR #73

**Pull request closed without merging**